src/System.Private.CoreLib/src/System/Diagnostics/StackFrameHelper.cs

```diff
@@ -34,6 +34,7 @@ internal class StackFrameHelper
 
         private IntPtr[] rgMethodHandle;
         private string[] rgAssemblyPath;
+        private Assembly[] rgAssembly;
```

Comment:
The existing `dynamicMethods` array serves pretty much same purpose. You could have just reused it, ie. if the method is dynamic, use it to store the dynamic method. Otherwise, if the assembly is unloadable, use it to store the loader allocator.

Also, the `dynamicMethods` array is passed around to other places.